api/v1alpha1/aws_clusterconfig_types.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/variables"
+)
+
+//+kubebuilder:object:root=true
+
+// AWSClusterConfig is the Schema for the awsclusterconfigs API.
+type AWSClusterConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AWSClusterConfigSpec `json:"spec,omitempty"`
+}
+
+// AWSClusterConfigSpec defines the desired state of AWSClusterConfig.
+type AWSClusterConfigSpec struct {
+	// +optional
+	Region *Region `json:"region,omitempty"`
+}
+
+func (AWSClusterConfigSpec) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "AWS Cluster configuration",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"region": Region("").VariableSchema().OpenAPIV3Schema,
+			},
+			Required: []string{"region"},
+		},
+	}
+}
+
+type Region string
+
+const (
```

Comment:
We should drop this. Neither upstream projects nor the aws-sdk-go have this sort of validation as far as i can tell.

Reply:
➕

Reply:
Aww i liked this :joy: Without this, if a dodgy region is specified then the cluster won't get rejected and we'd need to figure out how to surface the issue to the user.

Reply:
We could allow it to be configurable via the helm chart, and default to anything. For now though I think I'll just remove it and open an issue for future improvements.

Reply:
Removed. We can set this at the runtime hook setting in the clusterclass if need be. I'll work on that later, more as an example of how we can use settings than a necessary feature though.